service/create/service.go

```diff
@@ -517,26 +517,29 @@ func (s *Service) processCluster(cluster awstpr.CustomObject) error {
 		return microerror.Maskf(executionFailedError, fmt.Sprintf("could not retrieve host amazon account id: '%#v'", err))
 	}
 
-	// Create keypair.
-	var keyPair resources.ReusableResource
-	var keyPairCreated bool
-	{
-		var err error
-		keyPair = &awsresources.KeyPair{
-			ClusterName: key.ClusterID(cluster),
-			Provider:    awsresources.NewFSKeyPairProvider(s.pubKeyFile),
-			AWSEntity:   awsresources.AWSEntity{Clients: clients},
-		}
-		keyPairCreated, err = keyPair.CreateIfNotExists()
-		if err != nil {
-			return microerror.Maskf(executionFailedError, fmt.Sprintf("could not create keypair: '%#v'", err))
+	// An EC2 Keypair is needed for legacy clusters. New clusters provide SSH keys via cloud config.
```

Comment:
Conditional logic for whether to create a keypair based on if the version is set.